runtime.go

```diff
@@ -196,6 +196,8 @@ type Runtime struct {
 
 	Limiter *rate.Limiter
 	ticks   uint64
+
+	seenObjects map[*Object]bool
```

Comment:
should this be on the runtime struct or can we try to handle this on the Object struct itself?

Reply:
I'll give the Object struct a try! My thinking was that the current Object struct might change while running the function but the Runtime struct won't, but I can see how putting this in the Object struct would make more sense 😅